### PR TITLE
Debt ceiling per file

### LINKED ISSUE #12: debt_ceiling_per_file option to cap max debt for any one analyzed module

### PR DESCRIPTION
**Pull request by bglusman**
@SeanMarcia just to tag someone on this, i'll probably self merge unless you suggest any changes within an hour or so :-)

Fixes #12

### CODE REVIEW

**Comment by SeanMarcia** [comment]
reviewing now!

File: examples/.debt_ceiling.rb.example [hunk 1]
```diff
@@ -5,7 +5,7 @@ DebtCeiling.configure do |c|
   c.reduction_target = 100
   c.reduction_date   = 'Jan 1 2015'
   #set the multipliers per line of code in a file with each letter grade, these are the current defaults
-  c.grade_points = { a: 0, b: 10, c: 20, d: 40, f: 100 }
+  c.grade_points = { a: 0, b: 3, c: 13, d: 55, f: 144 }
```
Comment (SeanMarcia) [inline examples/.debt_ceiling.rb.example]: Did you pick fibonacci numbers for fun?

File: lib/debt_ceiling/debt.rb [hunk 2]
```diff
@@ -4,36 +4,54 @@ class Debt
     extend Forwardable
     DoNotWhitelistAndBlacklistSimulateneously = Class.new(StandardError)
 
-    attr_reader :file_attributes
     def_delegators :file_attributes, :path, :analysed_module, :module_name, :linecount, :source_code
-    def_delegator :analysed_module, :rating
-    attr_accessor :debt_amount
-    def_delegator :debt_amount, :to_i
+    def_delegator  :analysed_module, :rating
```
Comment (SeanMarcia) [inline lib/debt_ceiling/debt.rb]: why not add these to the delegators?

Comment (bglusman) [inline lib/debt_ceiling/debt.rb]: def_delegators is for assigning a bunch of methods to one object...  this is assigning rating to analysed_module, which is in turn delegated to file_attributes...  so either that's not an option, or I'm misunderstanding your question :-)

Comment (SeanMarcia) [inline lib/debt_ceiling/debt.rb]: You're right!

Comment (SeanMarcia) [inline lib/debt_ceiling/debt.rb]: You're right! I read too quickly and understood too slowly!

File: examples/.debt_ceiling.rb.example (hunk 1 above)
Comment (bglusman) [inline examples/.debt_ceiling.rb.example]: Yes, for fun, but also so there's some rhyme/reason to them?  Could just go with powers or 2 or something, this is a little weird as its every 3rd fibonacci number, but ...? seemed reasonable.

**Pull request merged**